### ISSUE TITLE
Suggested improvements to the temporal filter docs

### ISSUE THREAD

**Issue by chuck-alt-delete** [MaterializeInc/materialize#19789]
### Documentation request

- Remove the windowing idioms. My impression is these are idioms other streaming systems have to address their own limitations. Users typically want to just do a sliding window or bucketing with date_bin / trunc
- Remove delete timestamps. As far as I know, events aren't usually modeled with a delete timestamp and it makes the filter harder to understand
- remove grace period
- Include an "emit final value from a time window" example like [this](https://www.notion.so/materialize/Emit-Final-Result-of-Time-Window-e98b14b40032459b9577c1b3b6d017c4?pvs=4)

### Affected Pages

- https://materialize.com/docs/transform-data/patterns/temporal-filters/

### Related work

Slack convo:
- https://materializeinc.slack.com/archives/CPNFV9CVB/p1686149701031259

**Comment by chuck-alt-delete** [comment]
@jseldess I could take a shot at drafting a v1 for this with the revisions and a static image?

**Comment by jseldess** [comment]
@chuck-alt-delete, yes, please feel free to take a stab at this. Let me know if you'd like me to review a revision plan first.
